src/mdns/mdns_avahi.cpp

```diff
@@ -590,13 +636,47 @@ otbrError PublisherAvahi::UnpublishService(const char *aName, const char *aType)
 
 otbrError PublisherAvahi::PublishHost(const char *aName, const uint8_t *aAddress, uint8_t aAddressLength)
 {
-    OTBR_UNUSED_VARIABLE(aName);
-    OTBR_UNUSED_VARIABLE(aAddress);
-    OTBR_UNUSED_VARIABLE(aAddressLength);
+    otbrError    ret   = OTBR_ERROR_NONE;
+    int          error = 0;
+    char         fullHostName[kMaxSizeOfHost];
+    Host *       host;
+    AvahiAddress address;
 
-    VerifyOrDie(false, "PublishHost is not implemented with avahi");
+    VerifyOrExit(mState == State::kReady, errno = EAGAIN, ret = OTBR_ERROR_ERRNO);
+    VerifyOrExit(aName != nullptr, ret = OTBR_ERROR_INVALID_ARGS);
+    VerifyOrExit(aAddress != nullptr, ret = OTBR_ERROR_INVALID_ARGS);
+    VerifyOrExit(aAddressLength == sizeof(address.data.ipv6.address), ret = OTBR_ERROR_INVALID_ARGS);
 
-    return OTBR_ERROR_NONE;
+    SuccessOrExit(ret = MakeFullName(fullHostName, sizeof(fullHostName), aName));
+    SuccessOrExit(ret = GetOrCreateHost(mClient, aName, &host));
+    // Check if the update is needed
+    VerifyOrExit(!host->mIsCommitted || memcmp(host->mAddress.data.ipv6.address, aAddress, aAddressLength));
```

Comment:
Right, an attempt to call `avahi_entry_group_add_address` on an entry group which has already been committed results in a "Bad state" Avahi error. In theory, we could reset the entry group and commit it again every time `PublishHost` or `PublishService` is called, but it causes an unnecessary overhead on the Avahi side, so I thought it would be more elegant to introduce the "CommitHost" method which allows to do the commit explicitly. Let me know if you have a better idea on to handle this.

Reply:
What happens if I want to change my address for hostname `ot-my-host`? For example, `ot-my-host` was published with address `0xfd00::1` and I committed it. But later I want to change the address to `0xfd00::2` with another call to `Publisher::PublishHost()`, can we success?